### ISSUE TITLE
Broken code: 3.0 Spatial module failing on Near operator cannot be evaluated in this context.

### ISSUE THREAD

**Issue by ShakAttacked** [orientechnologies/orientdb#8048]
### OS: Centos

The following statement is giving, UnsupportedOperationException

`SELECT _id, $distance  FROM (SELECT FROM Location WHERE creator_id = "5a71b44511d82a4166670602" AND active = true ) WHERE [ latitude, longitude, $spatial ] NEAR [ 51.5074, 0.1278, { "maxDistance": 100 } ] ORDER BY $distance desc LIMIT 5 `

I'm copying the jar below into orientdb/lib.  The documentation doesn't say anything new?
http://central.maven.org/maven2/com/orientechnologies/orientdb-spatial/3.0.0RC1/orientdb-spatial-3.0.0RC1-dist.jar

Thanks.
Safwan

**Comment by luigidellaquila** [comment]
Hi @ShakAttacked 

Probably there is a problem in the implementation of the NEAR operator, I'll check it asap. Anyway, NEAR will be deprecated in the short term, my suggestion is to use `ST_Dwithin()` (http://orientdb.com/docs/3.0.x/indexing/Spatial-Index.html#stdwithin) or `ST_Contains()` instead

Thanks

Luigi

**Comment by ShakAttacked** [comment]
Thanks @luigidellaquila.  I've tested it and it's not working but I figured it out. After some research it seems that the documentation is out of date (or misleading as to what distance is).  I'll open a separate ticket for that but for those wondering why it's not working;  the query shows the various functions and what they return.  DistanceSphere will return in meters while ST_DWithin is in degrees.. which I cannot imagine is something needed as much as in meters.

**SELECT *,  ST_Distance(location, St_GeomFromText("POINT (-0.1277583 51.5073509)")) as distanceInDegrees, ST_Distance_Sphere(location, St_GeomFromText("POINT (-0.1277583 51.5073509)")) as distanceInMeters from Restaurant where ST_Distance_Sphere(location, St_GeomFromText("POINT (-0.1277583 51.5073509)")) < 2000**
